File: samples/wordnik-api-scala/src/main/scala/com/wordnik/client/AccountApi.scala
package com.wordnik.client

import com.wordnik.client.ApiInvoker
import com.wordnik.client.ApiException

import com.wordnik.client.model.AuthenticationToken
import com.wordnik.client.model.WordList
import com.wordnik.client.model.User
import java.util.Date
import com.wordnik.client.model.ApiTokenStatus
import scala.collection.mutable.HashMap

class AccountApi {
  var basePath: String = "http://api.wordnik.com/v4"
  var apiInvoker = ApiInvoker

  def authenticate (username: String, password: String) : Option[AuthenticationToken]= {
    // create path and map variables
    val path = "/account.{format}/authenticate/{username}".replaceAll("\\{format\\}","json").replaceAll("\\{" + "username" + "\\}",apiInvoker.escapeString(username))

    // query params
    val queryParams = new HashMap[String, String]
    val headerParams = new HashMap[String, String]

    // verify required params are set
    (Set(username, password) - null).size match {
       case 2 => // all required values set
       case _ => throw new Exception("missing required params")
    }
    if(String.valueOf(password) != "null") queryParams += "password" -> password.toString
    try {
      apiInvoker.invokeApi(basePath, path, "GET", queryParams.toMap, None, headerParams.toMap) match {
        case s: String =>
          Some(ApiInvoker.deserialize(s, "", classOf[AuthenticationToken]).asInstanceOf[AuthenticationToken])
        case _ => None
      }
    } catch {
      case ex: ApiException if ex.code == 404 => None
      case ex: ApiException => throw ex
    }
  }
  def authenticatePost (username: String, body: String) : Option[AuthenticationToken]= {
    // create path and map variables
    val path = "/account.{format}/authenticate/{username}".replaceAll("\\{format\\}","json").replaceAll("\\{" + "username" + "\\}",apiInvoker.escapeString(username))

    // query params
    val queryParams = new HashMap[String, String]
    val headerParams = new HashMap[String, String]

    // verify required params are set
    (Set(username, body) - null).size match {
       case 2 => // all required values set
       case _ => throw new Exception("missing required params")
    }
    try {
      apiInvoker.invokeApi(basePath, path, "POST", queryParams.toMap, body, headerParams.toMap) match {
        case s: String =>
          Some(ApiInvoker.deserialize(s, "", classOf[AuthenticationToken]).asInstanceOf[AuthenticationToken])
        case _ => None
      }
    } catch {
      case ex: ApiException if ex.code == 404 => None
      case ex: ApiException => throw ex
    }
  }
  def getWordListsForCurrentUser (auth_token: String, skip: Int, limit: Int) : Option[List[WordList]]= {
    // create path and map variables
    val path = "/account.{format}/wordLists".replaceAll("\\{format\\}","json")// query params
    val queryParams = new HashMap[String, String]
    val headerParams = new HashMap[String, String]

    // verify required params are set
    (Set(auth_token) - null).size match {
       case 1 => // all required values set
       case _ => throw new Exception("missing required params")
    }
    if(String.valueOf(skip) != "null") queryParams += "skip" -> skip.toString
    if(String.valueOf(limit) != "null") queryParams += "limit" -> limit.toString
    headerParams += "auth_token" -> auth_token
    try {
      apiInvoker.invokeApi(basePath, path, "GET", queryParams.toMap, None, headerParams.toMap) match {
        case s: String =>
          Some(ApiInvoker.deserialize(s, "List", classOf[WordList]).asInstanceOf[List[WordList]])
        case _ => None
      }
    } catch {
      case ex: ApiException if ex.code == 404 => None
      case ex: ApiException => throw ex
    }
  }
  def getApiTokenStatus (api_key: String) : Option[ApiTokenStatus]= {
    // create path and map variables
    val path = "/account.{format}/apiTokenStatus".replaceAll("\\{format\\}","json")// query params
    val queryParams = new HashMap[String, String]
    val headerParams = new HashMap[String, String]

    headerParams += "api_key" -> api_key
    try {
      apiInvoker.invokeApi(basePath, path, "GET", queryParams.toMap, None, headerParams.toMap) match {
        case s: String =>
          Some(ApiInvoker.deserialize(s, "", classOf[ApiTokenStatus]).asInstanceOf[ApiTokenStatus])
        case _ => None
      }
    } catch {
      case ex: ApiException if ex.code == 404 => None
      case ex: ApiException => throw ex
    }
  }
  def getLoggedInUser (api_key: String, auth_token: String) : Option[User]= {
    // create path and map variables
    val path = "/account.{format}/user".replaceAll("\\{format\\}","json")// query params
    val queryParams = new HashMap[String, String]
    val headerParams = new HashMap[String, String]

    // verify required params are set
    (Set(api_key, auth_token) - null).size match {
       case 2 => // all required values set
       case _ => throw new Exception("missing required params")
    }
    headerParams += "api_key" -> api_key
    headerParams += "auth_token" -> auth_token
    try {
      apiInvoker.invokeApi(basePath, path, "GET", queryParams.toMap, None, headerParams.toMap) match {
        case s: String =>
          Some(ApiInvoker.deserialize(s, "", classOf[User]).asInstanceOf[User])
        case _ => None
      }
    } catch {
      case ex: ApiException if ex.code == 404 => None
      case ex: ApiException => throw ex
    }
  }
  }

File: samples/wordnik-api-ruby/lib/Account_api.rb
require "uri"

class Account_api
  basePath = "http://api.wordnik.com/v4"
  # apiInvoker = APIInvoker

  def self.escapeString(string)
    URI.encode(string.to_s)
  end

  def self.authenticate (username,password,opts={})
    query_param_keys = [
      :password]
    
    # verify existence of params
    raise "username is required" if username.nil?
    raise "password is required" if password.nil?
    # set default values and merge with input
    options = {
      :username=>username,
    :password=>password,
    }.merge(opts)

    #resource path
    path = "/account.{format}/authenticate/{username}".sub('{format}','json').sub('{' + 'username' + '}', escapeString(username))
    

    
    # pull querystring keys from options
    queryopts = options.select do |key,value|
      query_param_keys.include? key
    end
    
    response = Swagger::Request.new(:GET, path, {:params=>queryopts,:headers=>nil, :body=>nil}).make.body
    AuthenticationToken.new(response)

  end


  def self.authenticate_post (username,body,opts={})
    query_param_keys = [
      ]
    
    # verify existence of params
    raise "username is required" if username.nil?
    raise "body is required" if body.nil?
    # set default values and merge with input
    options = {
      :username=>username,
    :body=>body,
    }.merge(opts)

    #resource path
    path = "/account.{format}/authenticate/{username}".sub('{format}','json').sub('{' + 'username' + '}', escapeString(username))
    

    
    # pull querystring keys from options
    queryopts = options.select do |key,value|
      query_param_keys.include? key
    end
    
    response = Swagger::Request.new(:POST, path, {:params=>queryopts,:headers=>nil, :body=>body.to_body}).make.body
    AuthenticationToken.new(response)

  end


  def self.get_word_lists_for_current_user (api_key,auth_token,skip,limit,opts={})
    query_param_keys = [
      :skip, :limit]
    
    # verify existence of params
    raise "api_key is required" if api_key.nil?
    raise "auth_token is required" if auth_token.nil?
    # set default values and merge with input
    options = {
      :api_key=>api_key,
    :auth_token=>auth_token,
    :skip=>skip,
    :limit=>limit,
    }.merge(opts)

    #resource path
    path = "/account.{format}/wordLists".sub('{format}','json')

    
    # pull querystring keys from options
    queryopts = options.select do |key,value|
      query_param_keys.include? key
    end
    
    headers = {
      api_key: api_key,
    }
    headers = {
      auth_token: auth_token,
    }
    response = Swagger::Request.new(:GET, path, {:params=>queryopts,:headers=>nil, :body=>nil}).make.body
    response.map {|response|WordList.new(response)}

  end


  def self.get_api_token_status (api_key,opts={})
    # set default values and merge with input
    options = {
      :api_key=>api_key,
    }.merge(opts)

    #resource path
    path = "/account.{format}/apiTokenStatus".sub('{format}','json')

    
    # pull querystring keys from options
    queryopts = options.select do |key,value|
      query_param_keys.include? key
    end
    
    headers = {
      api_key: api_key,
    }
    response = Swagger::Request.new(:GET, path, {:params=>queryopts,:headers=>nil, :body=>nil}).make.body
    ApiTokenStatus.new(response)

  end


  def self.get_logged_in_user (api_key,auth_token,opts={})
    query_param_keys = [
      ]
    
    # verify existence of params
    raise "api_key is required" if api_key.nil?
    raise "auth_token is required" if auth_token.nil?
    # set default values and merge with input
    options = {
      :api_key=>api_key,
    :auth_token=>auth_token,
    }.merge(opts)

    #resource path
    path = "/account.{format}/user".sub('{format}','json')

    
    # pull querystring keys from options
    queryopts = options.select do |key,value|
      query_param_keys.include? key
    end
    
    headers = {
      api_key: api_key,
    }
    headers = {
      auth_token: auth_token,
    }
    response = Swagger::Request.new(:GET, path, {:params=>queryopts,:headers=>nil, :body=>nil}).make.body
    User.new(response)

  end


  end
